### PR TITLE
Move charge counts to lower-right, to distinguish them from stack counts

### PR DESCRIPTION
**Pull request by SoyChai**
I suggested that item charges needed to be more distinct, and we agreed on discord that this was the best approach after a lengthy debate on whether to use colours instead.

image

The digits expand inward if larger numbers are needed.

### CODE REVIEW

**Comment by SomeZer0** [comment]
Would this be a toggle, or an outright change? I like where the current numbers are at, as it's more uniform.
I don't see why a change would even be required.

**Comment by devinfrench** [comment]
I originally moved them to the left corner so they would be uniform with stack count.

**Comment by SoyChai** [comment]
It's just UI design sense. It's a little odd to use the same number for two different meanings, right? Distinguishing the two like this reduces the mental effort required to understand what you're looking at in your inventory.

I do think this should be toggleable, honestly, but some older contributors disagreed. Any thoughts?

**Comment by SoyChai** [comment]
I imagine it's concern about falling into a slippery-slope where every tiny detail gets a config option. Though I think there's a strong case for this feature to be customizable, since our opinions on how to display it seem quite divided.

**Comment by SomeZer0** [comment]
>It's a little odd to use the same number for two different meanings, right? 

Would you also suggest different corners for ess pouches? Because "It's a little odd to use the same number for two different meanings, right?" Then we'd have a separate location for that overlay, because we can't have the same corner for stackables, or jewelry teleports as we do for ess pouches. Eventually, we'd run out of corners for different overlays.

>Distinguishing the two like this reduces the mental effort required to understand what you're looking at in your inventory.

I disagree. Either way you look at the item icon. You understand what you're looking at once you see the item icon. The positioning of the text doesn't make me confused over what item icon I'm looking at. Looking at an unnoted glory with a 3 in the top left corner does not make me think I have 3 noted glories, because it is obviously unnoted, based on the inventory icon. It's pretty obvious what the number represents, and what the item is.

I would actually argue the opposite of your point. Traditionally, in vanilla OSRS, counters that are displayed over items are in the top left.  It makes reasonable sense to assume to look there for any counters. Moving the counter to a different corner means that I have to use more mental effort, because I have to look in different corners for counters. This would also scale up for each and every counter we add a separate corner/location for.

If you have any specific examples of a counter(from the overlay) that is actually confusing, I'd like to see it.

>I do think this should be toggleable, honestly, but some older contributors disagreed. Any thoughts?

I would be opposed to an outright change, but I would not particularly care if a toggle is added.

**Comment by SoyChai** [comment]
> Would you also suggest different corners for ess pouches?

That's a good point that I hadn't thought about. I'd actually suggest the solution is to treat rune pouch contents as "charges" and merge them into this plugin, because the rune pouch is only ever one item, not a stack that can be split. That would feel consistent imo.

> Either way you look at the item icon. You understand what you're looking at once you see the item icon. [...] It's pretty obvious what the number represents, and what the item is.

This is easy for us to say as oldschool veterans ;)

I think that if Jagex implemented this into the game themselves, they would display it in a way which is easy for new players (who aren't aware of which items are stackable) to understand intuitively. I wouldn't hesitate to recommend RuneLite as a first client for a new player either.

I agree with what you're implying though: that veteran players should be prioritised, since they make up the vast majority of RuneLite's userbase, and they probably dislike unnecessary changes.

File: runelite-client/src/main/java/net/runelite/client/plugins/jewellerycount/JewelleryCountOverlay.java [hunk 1]
```diff
@@ -44,6 +36,15 @@
 import net.runelite.client.ui.overlay.components.TextComponent;
 import net.runelite.client.util.QueryRunner;
 
+import javax.inject.Inject;
+import java.awt.Dimension;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/jewellerycount/JewelleryCountOverlay.java]: Fix your imports

**Comment by deathbeam** [comment]
Closing as this is very old PR now and has a lot of merge conflicts, reopen if you want to re-implement it.

**Pull request closed without merging**